tools/cli/workflowCommands.go

```diff
@@ -1792,12 +1798,59 @@ func getResetEventIDByType(ctx context.Context, c *cli.Context, resetType, domai
 		if err != nil {
 			return
 		}
+	case "DecisionStartedTime":
+		earliestTime := parseTime(c.String(FlagEarliestTime), 0)
+		resetBaseRunID, decisionFinishID, err = getEarliestDecisionID(ctx, domain, wid, rid, earliestTime, frontendClient)
+		if err != nil {
+			return
+		}
 	default:
 		panic("not supported resetType")
 	}
 	return
 }
 
+func getEarliestDecisionID(ctx context.Context, domain string, wid string, rid string, earliestTime int64, frontendClient workflowserviceclient.Interface) (resetBaseRunID string, decisionFinishID int64, err error) {
+	resetBaseRunID = rid
+	req := &shared.GetWorkflowExecutionHistoryRequest{
+		Domain: common.StringPtr(domain),
+		Execution: &shared.WorkflowExecution{
+			WorkflowId: common.StringPtr(wid),
+			RunId:      common.StringPtr(rid),
+		},
+		MaximumPageSize: common.Int32Ptr(1000),
+		NextPageToken:   nil,
+	}
+
+	for {
+		resp, err := frontendClient.GetWorkflowExecutionHistory(ctx, req)
+		if err != nil {
+			return "", 0, printErrorAndReturn("GetWorkflowExecutionHistory failed", err)
+		}
+		found := false
+		for _, e := range resp.GetHistory().GetEvents() {
+			if e.GetEventType() == shared.EventTypeDecisionTaskCompleted {
+				if e.GetTimestamp() >= earliestTime {
+					found = true
+					decisionFinishID = e.GetEventId()
```

Comment:
nit: set decision eventID and break the outer loop directly instead of using `found`?